src/Http/Routing/src/DataTokensMetadata.cs

```diff
@@ -27,4 +29,16 @@ public DataTokensMetadata(IReadOnlyDictionary<string, object?> dataTokens)
     /// Get the data tokens.
     /// </summary>
     public IReadOnlyDictionary<string, object?> DataTokens { get; }
+
+    /// <inheritdoc/>
+    public override string ToString()
+    {
+        var metadataTokensBuilder = new StringBuilder();
+        foreach(var DataToken in DataTokens)
+        {
+            metadataTokensBuilder.AppendFormat(CultureInfo.InvariantCulture, "{0} ={1}", DataToken.Key, DataToken.Value);
+            metadataTokensBuilder.AppendLine();
+        }
+            return metadataTokensBuilder.ToString();
```

Comment:
I'd prefer comma limited rather than separate lines. Something like:

```cs
return string.Join(",", DataTokens.Select(t => $"{t.Key}={t.Value}"));
```